Getting started (#1805)
yewstack/yew: docs/getting-started/build-a-sample-app.md

title: Build a sample app
---

## Create Project

To get started, create a new cargo project.

```bash
cargo new yew-app
```

Open the newly created directory.

```bash
cd yew-app
```

> ## Run Hello World

Reviewer: Maybe
```suggestion
## Run a "Hello World" program
```
would be better?

Or possibly
```suggestion
## Run a hello world example
```

To verify the Rust environment is setup, run the initial project using the cargo build tool.  After output about the build process, you should see the expected "Hello World" message.

```bash
cargo run
```

> ## Convert project to Yew web application

Reviewer: Maybe?
```suggestion
## Converting the project into a Yew web application
```

yew = "0.17"
```

### Update main.rs

Replace the contents of `src/main.rs` with the following code.

> This template sets up your root `Component`, called `Model` which shows a button that updates itself when you click it.

Reviewer: ```suggestion
This template sets up a root `Component` called `Model` which renders a button that updates itself when clicked.
```

Take special note of `yew::start_app::<Model>()` inside `main()` which starts your app and mounts it to the page's `<body>` tag.
If you would like to start your application with any dynamic properties, you can instead use `yew::start_app_with_props::<Model>(..)`.

> 

Reviewer 2: I think the lines explaining the snippet should either go below (like it was before) or be in a more conversational tone. Example:

> We need to a template which sets up a root `Component` called `Model` which renders a button that updates itself when clicked.
> ...
> Now, let's replace the contents of `src/main.rs` with the following code.

Feel free to change the wording here

Author: Complete

</html>
```

## View your web applcation

> Run the following command to build your application, launch a local web hosting environment, and launch a web browser to view your application.

Reviewer: ```suggestion
Run the following command to build and serve the application locally.
```

```bash
trunk serve
```

> This will start a development server which continually updates the app every time you change something in your web application.

Reviewer: ```suggestion
Trunk will helpfully rebuild your application if you modify any of its files.
```

## Congratulations

You have now successfully setup your Yew development environment, and built your first web application.

> Experiment with this application and review the [examples](./examples.md) to further your learning.

Reviewer: Not really a problem otherwise.
```suggestion
Feel free to experiment with this application; you might find the [examples](./examples.md) useful.
```
